model/search.go

```diff
@@ -0,0 +1,123 @@
+package model
+
+import "github.com/ONSdigital/dp-renderer/model"
+
+// Search is the model struct for the cookies preferences form
+type SearchPage struct {
+	model.Page
+	Data       Search      `json:"data"`
+	Department *Department `json:"department"`
+}
+
+// Search represents all search parameters and response data of the search
+type Search struct {
+	Query      string           `json:"query"`
+	Filter     []string         `json:"filter,omitempty"`
+	Sort       Sort             `json:"sort,omitempty"`
+	Pagination model.Pagination `json:"pagination,omitempty"`
+	Response   Response         `json:"response"`
+}
+
+// Sort represents all the information of sorting related to the search page
+type Sort struct {
+	Query              string        `json:"query,omitempty"`
+	LocaliseFilterKeys []string      `json:"filter_text,omitempty"`
+	LocaliseSortKey    string        `json:"sort_text,omitempty"`
+	Options            []SortOptions `json:"options,omitempty"`
+}
+
+// SortOptions represents all the information of different sorts available
+type SortOptions struct {
+	Query           string `json:"query,omitempty"`
+	LocaliseKeyName string `json:"localise_key"`
+}
+
+// Response represents the search results
+type Response struct {
+	Count                 int           `json:"count"`
+	Categories            []Category    `json:"categories"`
+	Items                 []ContentItem `json:"items"`
+	Suggestions           []string      `json:"suggestions,omitempty"`
+	AdditionalSuggestions []string      `json:"additional_suggestions,omitempty"`
+}
+
+// Category represents all the search categories in search page
+type Category struct {
+	Count           int           `json:"count"`
+	LocaliseKeyName string        `json:"localise_key"`
+	ContentTypes    []ContentType `json:"content_types"`
+}
+
+// ContentType represents the type of the search results and the number of results for each type
+type ContentType struct {
+	Type            string `json:"type"`
+	Count           int    `json:"count"`
+	LocaliseKeyName string `json:"localise_key"`
+}
+
+// ContentItem represents each search result
+type ContentItem struct {
+	Description Description `json:"description"`
+	Type        string      `json:"type"`
+	URI         string      `json:"uri"`
+	Matches     *Matches    `json:"matches,omitempty"`
```

Comment:
Could we replace this with `MatchDescriptions` and remove a level of nesting, as the `Matches` struct seems to only include that?